ImperatorToCk2/Characters.java

```diff
@@ -0,0 +1,208 @@
+                            output[14] = qaaa.split("=")[1];
+                            output[14] = output[14].substring(2,output[14].length()-2); 
+
+                            try {
+                                if (output[14].split(" ")[1] != null) {
+                                    output[14] = output[14].split(" ")[output[14].split(" ").length-1];   
+                                }
+                            }catch (java.lang.ArrayIndexOutOfBoundsException exception) {
+
+                            }
+                        }
+                        else if (qaaa.split("=")[0].equals( tab+"children" ) ) {
+
+                
+                            output[15] = qaaa.split("=")[1];
+                            output[15] = output[15].substring(2,output[15].length()-2);    
+
+                        }
+
+                        //popTotal
+                        else if (qaaa.split("=")[0].equals( tab+"age" ) ) {
+                            aqq = aqq + 1;
+                            output[3] = qaaa.split("=")[1];
+                        }
+
+                        //might be used or ignored
+                        else if (qaaa.equals( tab+"female=yes" ) ) {
+                            aqq = aqq + 1;
+                            output[4] = "f";
+                        }
+
+                        else if (qaaa.split("=")[0].equals( tab+"death_date" ) ) {
+                            aqq = aqq + 1;
+                            flag = 1; //end loop, babies which die don't have any experience field
+                            output[4] = output[4]+"_"+qaaa.split("=")[1];
+                            output[6] = "0";
+                        }
+
+                        else if (qaaa.split("=")[0].equals( tab+"character_experience" ) ) {
+                            aqq = aqq + 1;
+                            flag = 1; //end loop
+                            output[6] = qaaa.split("=")[1];
+                        }
+
+                    }
+
+                }
+            }
+
+        }catch (java.util.NoSuchElementException exception){
+            endOrNot = false;
+
+        }   
+
+        return output;
+    
+    }
+    public static String importAndConvDynasty (String dir, String IDnum, String backupName, String file) throws IOException
+    {
+
+        String tab = "	";
+
+        FileInputStream fileIn= new FileInputStream(file);
+        Scanner scnr= new Scanner(fileIn);
+
+        int flag = 0;
+
+        String keyWord = tab+tab+IDnum;
+
+        int aqq = 0;
+        boolean endOrNot = true;
+        String vmm = scnr.nextLine();
+        String qaaa = vmm;
+        String output = "noName"; //default for no name
+
+      
+        try {
+            while (endOrNot = true){
+
+                qaaa = scnr.nextLine();
+                if (qaaa.split("=")[0].equals(keyWord)){
+                    endOrNot = false;
+                    while (flag == 0) {
```

Comment:
You can make this
```
while (true) { 
```
Then when you're ready to stop, instead of setting the flag, just give the statement
```
break;
```